### PR TITLE
docs: Fix conflicting statements about initialization in developer notes

### PR DESCRIPTION
**Pull request by practicalswift**
Fix conflicting statements about initialization in developer notes.

Context: https://github.com/bitcoin/bitcoin/pull/12785#issuecomment-378941151

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -450,12 +450,17 @@ C++ data structures
 
 - Vector bounds checking is only enabled in debug mode. Do not rely on it
 
-- Make sure that constructors initialize all fields. If this is skipped for a
```
Comment (laanwj) [inline doc/developer-notes.md]: Please don't remove this part about optimization.

**Comment by practicalswift** [comment]
@laanwj Updated. Please re-review :-)

**Comment by promag** [comment]
ACK b119e78, looks good.

**Comment by Wenkai-Yang** [comment]
mute the thread发自我的华为手机-------- 原始邮件 --------主题：Re: [bitcoin/bitcoin] docs: Fix conflicting statements about initialization in developer notes (#12896)发件人："Wladimir J. van der Laan" 收件人：bitcoin/bitcoin 抄送：Subscribed Merged #12896.

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/bitcoin/bitcoin","title":"bitcoin/bitcoin","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/bitcoin/bitcoin"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"Merged #12896."}],"action":{"name":"View Pull Request","url":"https://github.com/bitcoin/bitcoin/pull/12896#event-1562216246"}}}

**Pull request merged**